### ISSUE TITLE
`[pydoclint]` DOC501: False positive for exception created via classmethod

### ISSUE THREAD

**Issue by mschoettle** [astral-sh/ruff#23570]
### Summary

I am raising a custom Pydantic `ValidationError` via `ValidationError.from_exception_data` and the rule does not detect that this is `ValidationError`:

```console
Raised exception `from_exception_data` missing from docstring (DOC501) [Ln 6, Col 5]
Raised exception is not explicitly raised: `ValidationError` (DOC502) [Ln 6, Col 5]
```

Playground: https://play.ruff.rs/b36f20cd-c123-4090-a9bd-d268cc89cecf

### Version

ruff 0.15.2

**Comment by ntBre** [comment]
Thanks for the report! I wonder if we could try resolving the exception name from the docstring while parsing here:

https://github.com/astral-sh/ruff/blob/728609a06e590e8b458714b4e17d6b16828c21e7/crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs#L772-L774

I think what's happening is that we turn the `ValidationError` in the docstring into just a `ValidationError` qualified name, and then the real `ValidationError` binding is `pydantic.ValidationError`. The rule works around this mismatch by checking if the _ends_ of the qualified names overlap:

https://github.com/astral-sh/ruff/blob/728609a06e590e8b458714b4e17d6b16828c21e7/crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs#L1343-L1346 

rather than comparing the start, which would discard the trailing method call.
